src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Exceptions;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.EditVacancyInfo;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using SFA.DAS.Http;
+using SFA.DAS.Http.TokenGenerators;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Services.ProviderRelationship
+{
+    public class ProviderRelationshipsService : IProviderRelationshipsService
+    {
+        private readonly ProviderRelationshipApiConfiguration _configuration;
+        private readonly ILogger<ProviderRelationshipApiConfiguration> _logger;
+        private readonly IJobsVacancyClient _jobsVacancyClient;
+
+        public ProviderRelationshipsService(IOptions<ProviderRelationshipApiConfiguration> configuration,
+            ILogger<ProviderRelationshipApiConfiguration> logger,
+            IJobsVacancyClient jobsVacancyClient)
+        {
+            _configuration = configuration.Value;
+            _logger = logger;
+            _jobsVacancyClient = jobsVacancyClient;
+        }
+
+        public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(long ukprn)
+        {
+            IEnumerable<EmployerInfo> employerInfos = new List<EmployerInfo>();
+            var httpClient = CreateHttpClient(_configuration);
+            var queryData = new { Ukprn = ukprn, Operation = "Recruitment" };            
+            var uri = new Uri(AddQueryString("accountproviderlegalentities", queryData), UriKind.RelativeOrAbsolute);
+
+            try
+            {
+                var response = await httpClient.GetAsync(uri);
+                if (response.IsSuccessStatusCode)
+                {
+                    var content  = await response.Content.ReadAsStringAsync();
```

Comment:
Worth using RestSharp? We already use it for BankHolidayUpdateService.

Reply:
Any particular reason for using RestSharp over Http Client, I don't mind either way.

Reply:
Not really, Just personal preference.

Reply:
> Don’t use restsharp please just leave as it is

reason being?

Reply:
My take on it.  Our efforts should be spent writing value-added code.  Serializing/deserializing json data adds no value.  If there is library that will do that for us and reduce the code we need to write then we should consider it.  The moment it adds complexity we should ditch it.

Reply:
I kind of agree with Sachin to keep external dependency down as much as possible. I would have been inclined to use RestSharp if we weren't dealing with json elsewhere, but we do, so I am leaving this as is for now.

Reply:
Sounds good to me.